CMakeLists.txt

```diff
@@ -1,26 +1,26 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.16)
 
 project(FluxHarmonic C)
 
 find_package(PkgConfig REQUIRED)
-
-#########################################
-# flux-compose executable
-
-add_executable(flux-compose src/core/main.c src/core/repl.c)
-target_link_libraries(flux-compose PUBLIC flux)
-
-#########################################
-# flux shared library
 find_package(Threads REQUIRED)
 
 # We could also use find_package(SDL), but that's only CMake 3.19
 pkg_check_modules(SDL2 REQUIRED IMPORTED_TARGET sdl2)
 pkg_check_modules(SDL2-Gfx REQUIRED IMPORTED_TARGET SDL2_gfx)
 pkg_check_modules(SDL2-Image REQUIRED IMPORTED_TARGET SDL2_image)
 
-add_library(flux SHARED lib/flux.h lib/lib.c lib/log.c lib/scene.c lib/mem.c lib/file.c lib/script.c lib/vector.c)
-target_link_libraries(flux PkgConfig::SDL2 PkgConfig::SDL2-Gfx PkgConfig::SDL2-Image Threads::Threads)
+include_directories(lib src)
+
+add_library(flux)
+target_link_libraries(flux PkgConfig::SDL2 PkgConfig::SDL2-Gfx
+  PkgConfig::SDL2-Image Threads::Threads)
+
+add_executable(flux-compose)
+target_link_libraries(flux-compose flux)
+
+add_subdirectory(lib)
```

Comment:
What's the difference between this and `include_directories`?

Reply:
`include_directories()` adds the directory to the include path (i.e. -Ilib). As for `add_subdirectory()` it adds the directory to as a build unit, i.e. it tells make to enter the subdirectory and look for other sources there.

Reply:
Ahhh, that makes sense.  Thanks a bunch!